### PR TITLE
Add API to send confirmation email

### PR DESCRIPTION
**Pull request by SomTambe**
Addressing issue #34 . I added an API which can be used for any mailing related task, not only to send confirmation email.

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+from campusdiscussbackend.settings import EMAIL_HOST_USER
+# @csrf_exempt 
+# Do not use until you want it to POST
 
 # Create your views here.
+
+def send_email(request):
+    if request.method == "POST":
+        emailid=JSONParser().parse(request)
+        serialiser=ActivationMailer(emailid)
```
Comment (dryairship) [inline users/views.py]: Nit-picking: `serializer`.

File: users/views.py [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+        subject="Activation mail"
```
Comment (dryairship) [inline users/views.py]: You should take subject and body from the settings file perhaps. @ditttu ? @yashsriv ?

If you intend to make a generic mailer function, not just for activation emails, then you should take the subject and body as parameters. Then call that function from the request handler function.

File: users/views.py [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+        body="Body"
+        recepient=serialiser.data['email']
```
Comment (dryairship) [inline users/views.py]: Nitpicking: `recipient`.

File: users/views.py [hunk 4]
```diff
@@ -1,3 +1,23 @@
...
+        send_email(subject, body, EMAIL_HOST_USER, [recepient], fail_silently=False)
+        return redirect("https://redirect.link")
```
Comment (dryairship) [inline users/views.py]: The redirect URL should definitely be taken from the settings file.

**Review by dryairship — COMMENTED** [review]
We should also perhaps check whether or not we have already mailed a verification code to this user. The API can be misused without this check. 

Perhaps we should store a timestamp of the last time we sent a verification mail to this user, and send another mail only if the it has been > `someFixedTimePeriod` since the last time.

Since there are so many settings for email (host,port,user,port,subject,body...), perhaps we can make an entirely new settings file for just email settings eg: `settings_email.py` and load it in our main settings file using `from .settings_email import *`.

File: users/views.py (hunk 1 above)
Comment (dryairship) [reply users/views.py]: Also, we should stick to snake case for variables. Therefore, use `email_id`.

Comment (SomTambe) [reply users/views.py]: I thought we were sending the activation link itself in the mail. That is the reason I did not think of verification code. Could you please confirm @dryairship ?

**Comment by SomTambe** [comment]
I have added a settings_email.py file as said. Now we can directly create purpose-wise email classes in it, and pass them as parameters in the send_email function. Kindly review.

File: users/views.py [hunk 5]
```diff
@@ -4,20 +4,25 @@
...
 
 # Create your views here.
-
-def send_email(request):
+'''
+The send_email function is an API for sending any email.
+In place of email_class, you can type in the class of email-type as
+mentioned in settings_email.py. An example class is given as ExampleEmail
+'''
+def send_email(request, email_class):
```
Comment (dryairship) [inline users/views.py:17]: Are you sure this even works?

This is an API function. We don't need an API function to send an email - that should be done using a utility function.  The APIs (like sending a verification mail API) which require using that utility should call that utility function.

See [this question](https://stackoverflow.com/questions/58730090/location-for-app-wide-utils-file-in-a-django-project) and the link in the accepted answer.

**Comment by SomTambe** [comment]
I will make all the required changes, put properly designed code. Sorry for bad planning

**Pull request closed without merging**